src/datadoc/frontend/callbacks/dataset.py

```diff
@@ -287,18 +283,26 @@ def accept_dataset_metadata_date_input(
     )
 
 
-def dataset_metadata_control() -> dbc.Alert | None:
-    """Check obligatory metadata values for dataset."""
-    missing_metadata: list = []
-    dataset = state.metadata.dataset
-    for field in dataset:
-        if not obligatory_metadata(field, OBLIGATORY_DATASET_METADATA_IDENTIFIERS):
-            field_name = get_metadata_field_display_name(
-                field,
-                OBLIGATORY_DATASET_METADATA_IDENTIFIERS_AND_DISPLAY_NAME,
+def dataset_control(error_message: str | None) -> dbc.Alert | None:
+    """Check obligatory metadata values for dataset.
+
+    Args:
+        error_message(str): A message generated by ObligatoryDatasetWarning containing names of fields missing value.
+    """
+    missing_metadata = [
+        f[1]
+        for f in OBLIGATORY_DATASET_METADATA_IDENTIFIERS_AND_DISPLAY_NAME
+        if (
+            (
+                f[1] == "Beskrivelse"
+                and error_message
+                # and r"\bdescription\b" in error_message
+                and re.search(r"\bdescription\b", error_message)
```

Comment:
It's not obvious why description needs to be a special case, could you add a comment explaining that?

Reply:
Added a generic method for all fields